specification/web/resource-manager/Microsoft.Web/2016-03-01/Diagnostics.json

```diff
@@ -0,0 +1,1512 @@
+          "description": "End time of the corelated event",
+          "type": "string"
+        },
+        "message": {
+          "description": "Message describing the event",
+          "type": "string"
+        },
+        "source": {
+          "description": "Represents the name of the Detector",
+          "type": "string"
+        },
+        "priority": {
+          "format": "double",
+          "description": "Represents the rank of the Detector",
+          "type": "number"
+        },
+        "metaData": {
+          "description": "Downtime metadata",
+          "type": "array",
+          "items": {
+            "type": "array",
+            "items": {
+              "$ref": "./CommonDefinitions.json#/definitions/NameValuePair"
+            }
+          }
+        },
+        "type": {
+          "description": "Represents the type of the Detector",
+          "enum": [
+            "ServiceIncident",
+            "AppDeployment",
+            "AppCrash",
+            "RuntimeIssueDetected",
+            "AseDeployment",
+            "UserIssue",
+            "PlatformIssue",
+            "Other"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "IssueType",
+            "modelAsString": false
+          }
+        },
+        "solutions": {
+          "description": "List of proposed solutions",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Solution"
+          }
+        }
+      }
+    },
+    "DetectorDefinition": {
+      "description": "Class representing detector definition",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "./CommonDefinitions.json#/definitions/ProxyOnlyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "DetectorDefinition resource specific properties",
+          "properties": {
+            "name": {
+              "description": "Name of the detector",
+              "type": "string",
+              "readOnly": true
+            },
+            "displayName": {
+              "description": "Display name of the detector",
+              "type": "string",
+              "readOnly": true
+            },
+            "description": {
+              "description": "Description of the detector",
+              "type": "string",
+              "readOnly": true
+            },
+            "rank": {
+              "format": "double",
+              "description": "Detector Rank",
+              "type": "number",
+              "readOnly": true
+            },
+            "isEnabled": {
+              "description": "Flag representing whether detector is enabled or not.",
+              "type": "boolean",
+              "readOnly": true
+            }
+          },
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "DiagnosticAnalysis": {
+      "description": "Class representing a diagnostic analysis done on an application",
```

Comment:
@ravbhatnagar I have added the ProxyOnlyResource to this class as well as DiagnosticDetectorResponse, per your comment. However, I am not sure what to do about the id, name, and type missing. These are fields that are there in the response but it doesn't seem that the way that we generate the swagger adds them.

Reply:
Ok, please explore how you can fix it on your side in the auto generated swagger. Approving this for now since the service returns these properties.